.github/workflows/update-docker-staging.yml

```diff
@@ -1,7 +1,9 @@
-name: Run on Push
+name: Build and Publish Docker on Staging
 
 on:
   push:
+    branches:
+      - staging
```

Comment:
The staging Docker image is updated after each PR merge. This branch serves as a caching and testing pool before changes are merged into main, which basically means the code goes live.